### ISSUE TITLE
[Question / Issue?] NSOperation Serial Queue + GDCoreDataConcurrencyDebugging = false positive?

### ISSUE THREAD

**Issue by Soph** [GrahamDennis/GDCoreDataConcurrencyDebugging#11]
First of all: Thanks for the project!

It helped me finding at least one issue im my code yet and I guess there are more since I haven't tested all aspects of that app. 

But right now I'm struggling with something which I think (since a few minutes) isn't an issue of my code but more an issue with the implementation of ValidateConcurrency. 

I'm using NSURLSession in my app. For the background downloading stuff it needs a serial NSOperationQueue where it will make all callbacks. I have a dedicated NSManagedObjectContext (NSConfinementConcurrencyType) for that Queue. 

Right now I get lots of warnings from GDCoreDataConcurrencyDebugging that I tried to understand or solve but couldn't find any issues in my code. But I guess I know the reason now: The NSOperationQueue in serial mode does always use only one thread but not always the same thread. So my context is created when needed (in the getter) and concurrencyIdentifier is set to that thread/dispatch_queue which the NSOperationQueue is using at that time. Sometime during the runtime of the app (my guess is when the NSOperationQueue is empty for a certain period of time) the dispatch_queue changes and so ValidateConcurrency returns NO.

Does that make sense, would you agree with my findings or do you see any other issues?

So basically instead of only looking at the current dispatch queue for a context with NSConfinementConcurrencyType we could also check if the context is inside a NSOperationQueue ([NSOperationQueue currentQueue]) and if this is a serial queue (maxConcurrentOperationCount == 1) and then use the NSOperationQueue as the concurrencyIdentifier. What do you think?
